src/meltano/core/plugin_lock_service.py

```diff
@@ -61,7 +61,7 @@ def save(self) -> None:
         locked_def = StandalonePlugin.from_variant(self.variant, self.definition)
 
         with self.path.open("w") as lockfile:
-            json.dump(locked_def.canonical(), lockfile, indent=2)
+            lockfile.write(json.dumps(locked_def.canonical(), indent=2) + "\n")
```

Comment:
This might have slightly better performance

```suggestion
            json.dump(locked_def.canonical(), lockfile, indent=2)
            lockfile.write("\n")
```

But how it was before is fine too.